src/serdes/binary-serializer.ts

```diff
@@ -0,0 +1,107 @@
+import { strict as assert } from 'assert'
+import { parseBytes, bytesToHex } from '../utils/bytes-utils'
+import { makeClass } from '../utils/make-class'
+import { Enums } from '../enums'
+
+const BytesSink = {
```

Comment:
This also strikes me as confusing. 

I'm not sure what is going on here or what the purpose is?